packages/ibm-products/src/components/OptionsTile/OptionsTile.stories.js

```diff
@@ -141,6 +141,7 @@ const TemplateStatic = ({ enabled, ...rest }) => {
     <OptionsTile
       onToggle={onToggle}
       onChange={onChange}
+      onClick={onToggle}
```

Comment:
The `OptionsTile` doesn't accept an `onClick` prop which means this will be spread to the outer `<div>` which would be invalid without additional semantics and creates an unintended interaction.

```suggestion

```

Reply:
Im not clear what do you mean by unintended interactions? And why is onClick not accepted when Toggle component itself has an onClick?

Reply:
The `OptionsTile` component does not have a prop called `onClick`. This means when you pass it anyways, it will be emitted on the outer `<div>` with any other attributes the component doesn't handle. This happens here: https://github.com/carbon-design-system/ibm-products/blob/40fe19b4ef32f97f01e8664bbe81fc1e459d0f3f/packages/ibm-products/src/components/OptionsTile/OptionsTile.js#L243-L244

The interaction pattern for OptionsTile is that clicking on the header will open and close the content of the tile. To enable or disable the tile, the user must click on the toggle directly. Otherwise they wouldn't be able to open or close the tile without unexpected side effects that may come from enabling or disabling an option.

Have you tried my suggestion to update `/packages/ibm-products/src/components/OptionsTile/OptionsTile.js` with the code shared above?

Reply:
Did not work for me, can you please check the sandbox link which i have commented

Reply:
https://codesandbox.io/p/sandbox/zen-kapitsa-wqx572

Reply:
In your linked codesandbox, you can see how adding the `onClick` is creating the unwanted interaction I was mentioning. Since it is emitted on the outer `<div>`, whenever you click anywhere on the OptionsTile it toggles between enabled and disabled.

https://github.com/carbon-design-system/ibm-products/assets/28265588/a5a46ec8-cfdc-4030-9db5-73b40ce28131

Instead, it should only switch between enabled and disabled when the user clicks the toggle on the right.
Right now, the `<OptionsTile>` implementation wrongly sets the `<Toggle>` props `labelA` and `labelB` to empty strings. This is the root cause of the issue reported in #3163.

The reason behind this is that the Carbon `<Toggle>` uses a `<button>` under the hood which is set to be visually hidden. Clicking on the generated `<label>` element will trigger the button click and therefore the `onToggle` event. Since the `<OptionsTile>` currently sets the label to empty strings though, there is no clickable target left.

In order to fix this, the `<OptionsTile>` should use `props.hideLabel` for the `<Toggle>` instead as this will influence the underlying markup and cause the `<Toggle>` to render without labels but with a clickable target.

The fix needs to be applied to `/packages/ibm-products/src/components/OptionsTile/OptionsTile.js`:
https://github.com/carbon-design-system/ibm-products/blob/40fe19b4ef32f97f01e8664bbe81fc1e459d0f3f/packages/ibm-products/src/components/OptionsTile/OptionsTile.js#L260-L271
```diff
<Toggle
  id={`${id}-toggle`}
  className={`${blockClass}__toggle`}
  toggled={enabled}
- labelA=""
- labelB=""
  aria-labelledby={titleId}
+ hideLabel
  onToggle={onToggle}
  size="sm"
  disabled={isLocked}
- labelText={title}
/>
```